app/views/settings/edit.html.erb

```diff
@@ -1,21 +1,38 @@
 <% @title = t(".title") %>
-
 <div class="t-body">
   <h2><%= t ".mfa.multifactor_auth" %></h2>
   <% if @user.mfa_enabled? %>
     <p><%= t(".mfa.enabled_html") %></p>
     <%= form_tag multifactor_auth_path, method: :put, id: "mfa-edit" do %>
       <%= label_tag :level, t(".mfa.level.title"), class: "form__label" %>
-      <%= select_tag :level, options_for_select([
-        [t(".mfa.level.ui_and_api"), "ui_and_api"],
-        [t(".mfa.level.ui_and_gem_signin"), "ui_and_gem_signin"],
-        [t(".mfa.level.ui_only"), "ui_only"],
-        [t(".mfa.level.disabled"), "disabled"]], @user.mfa_level), class: "form__input form__select" %>
+      <%# Note: Scrappy version ... this can obviously be DRY'd up %>
+      <% if @user.owner_of_most_downloaded_gem? %>
+         <%= select_tag :level, options_for_select([
+          [t(".mfa.level.ui_and_api"), "ui_and_api"],
+          [t(".mfa.level.ui_and_gem_signin"), "ui_and_gem_signin"],
+          [t(".mfa.level.ui_only"), "ui_only"],
+          ["Deregister device", "disabled"]], @user.mfa_level), class: "form__input form__select" %>
```

Comment:
It's great to see a different drop down with the available options `ui and api`, `ui and gem signin` and `deregister a device`! I think we should also add some validation in the backend to make sure nobody is trying to set their level to `ui_only` by modifying the DOM.

Reply:
Addressed in https://github.com/Shopify/rubygems.org/pull/2/commits/394bf77f7672e42b3df73776b165f2c5e9d6bc52